### PR TITLE
rename CUDAWatchpoint to GPUWatchpoint, fix header include

### PR DESCRIPTION
**Pull request by dtaller**
I need to include Cuda watch point for both the hip and cuda debug build, not just the cuda build as before

Also rename CudaWatchpoint to GPUWatchpoint

### CODE REVIEW

File: src/care/util.h [hunk 1]
```diff
@@ -15,6 +15,9 @@
 // Other library headers
 #if defined(__CUDACC__)
 #include "cuda.h"
+#endif
+
+#if defined(CARE_GPUCC)
 #if defined(CARE_DEBUG)
 #include "care/CUDAWatchpoint.h"
```
Comment (adayton1) [inline src/care/util.h:22]: Should this file be renamed GPUWatchpoint.h? Does it have both HIP and CUDA code in it?

Comment (dtaller) [reply src/care/util.h:22]: That's a good point. Yes, it is designed for both. Will do

Comment (dtaller) [reply src/care/util.h]: Done.

**Review by adayton1 — APPROVED** [review]
LGTM

**Pull request merged**